WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -48,6 +48,8 @@ final class DashboardViewModel: ObservableObject {
         dashboardCards.filter { $0.availability == .show && $0.enabled }
     }
 
+    private var savedCards: [DashboardCard] = []
```

Comment:
I am concerned about adding a new property to save the cards. We also already have `var dashboardCards: [DashboardCard]`, and the new `savedCards` which will make the code harder to follow and debug. 

From the PR description, I think that we do this to "Reduce amount of time saved cards are loaded.". Was the previous logic causing any bugs? Would it be possible to explore another solution without introducing the `savedCards` property?

Reply:
Suggestion: If the requirement is that the locally `savedCards` values to be ready in order to render the cards on screen, I think we should wait for the value to be available (instead of running a `Task` in `DashboardViewModel` init) and show a loading indicator if needed. This is because we can't be sure that the `Task` started in `init` will run/finish before we access the `savedCards` value. What do you think?

Reply:
>  "Reduce amount of time saved cards are loaded.". Was the previous logic causing any bugs? Would it be possible to explore another solution without introducing the savedCards property?

So the issue I'm seeing when debugging is that this line:

```
let savedCards = await loadDashboardCards() ?? []
```

https://github.com/woocommerce/woocommerce-ios/pull/12899/files#diff-5e4b4001de415046921cf84e4ce2b6c16646ae1bb2067aa6d9ec1fbfa2a855e5L522

is called over and over even in just one pull-to-refresh action. I think it is because it's part of the combine in `refreshDashboardCard()`, so it's re-called each time a publisher is updated. However, the value of the saved cards do not change during pull-to-refresh, so it seems inefficient to me to load it multiple times only to get the same value each time.

Reply:
Decided to revert the part about savedCards refactoring so this discussion can be resolved for now.